sdk/src/main/java/io/customer/sdk/queue/QueueStorage.kt

```diff
@@ -104,6 +105,23 @@ internal class QueueStorageImpl internal constructor(
         return jsonAdapter.fromJsonOrNull(fileContents)
     }
 
+    override fun deleteGroup(groupStartTask: String): List<QueueTaskMetadata> {
+        val inventory = getInventory()
+
+        // find the start of the group.
+        val groupStart = inventory.filter { it.groupStart == groupStartTask }
```

Comment:
I might have found an edge case that could cause a bug. 

If the caller of `deleteGroup` ever calls `delete(taskStorageId:)` before calling `deleteGroup`, this might cause a bug since the task `groupStart` will no longer exist. 

I think it would be good for this function to handle that edge case and this could be another reason for encapsulating the delete task + delete group members into this class could be a good idea.